src/PubSub/PubSubClient.php

```diff
@@ -0,0 +1,308 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\PubSub;
+
+use Google\Cloud\ClientInterface;
+use Google\Cloud\ClientTrait;
+use Google\Cloud\PubSub\Connection\Rest;
+
+/**
+ * Google Cloud Pub/Sub client. Allows you to send and receive
+ * messages between independent applications. Find more information at
+ * [Google Cloud Pub/Sub docs](https://cloud.google.com/pubsub/docs/).
+ */
+class PubSubClient implements ClientInterface
+{
+    use ClientTrait;
+    use ResourceNameTrait;
+
+    const FULL_CONTROL_SCOPE = 'https://www.googleapis.com/auth/pubsub';
+
+    /**
+     * @var array  Any configuration options
+     */
+    private $config;
+
+    /**
+     * @var string The project ID created in the Google Developers Console.
+     */
+    private $projectId;
+
+    /**
+     * Create a PubSub client.
+     *
+     * Example:
```

Comment:
My bad, there is [inline @example](https://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_tags.inlineexample.pkg.html) as well but looks clunky.